src/threadprofiler.js

```diff
@@ -24,24 +24,34 @@ var emscriptenThreadProfiler = {
   },
 
   updateUi: function updateUi() {
-    if (typeof PThread === 'undefined') return; // Likely running threadprofiler on a singlethreaded build, or not initialized yet, ignore updating.
+    if (typeof PThread === 'undefined') {
+      // Likely running threadprofiler on a singlethreaded build, or not
+      // initialized yet, ignore updating.
+      return;
+    }
     var str = '';
     var mainThread = _emscripten_main_browser_thread_id();
 
     var threads = [mainThread];
-    for(var t in PThread.pthreads) threads.push(PThread.pthreads[t].threadInfoStruct);
+    for (var i in PThread.pthreads) {
+      threads.push(PThread.pthreads[i].threadInfoStruct);
+    }
 
-    for(var i = 0; i < threads.length; ++i) {
-      var threadPtr = threads[i];//(t == mainThread ? mainThread : maiPThread.pthreads[t].threadInfoStruct;
-      var profilerBlock = Atomics.load(HEAPU32, (threadPtr + 20 /*C_STRUCTS.pthread.profilerBlock*/ ) >> 2);
+    for (var i = 0; i < threads.length; ++i) {
+      var threadPtr = threads[i];
+      var profilerBlock = Atomics.load(HEAPU32, (threadPtr + 8 /* {{{ C_STRUCTS.pthread.profilerBlock }}}*/) >> 2);
```

Comment:
does this file not get the preprocessor run on it, which would let the `{{{ }}}` stuff work?

Reply:
I tried that and it didn't work.   I'm a little confused about that does and doesn't get pre-processed but this file certainly doesn't seem to.